src/auth/index.ts

```diff
@@ -0,0 +1,62 @@
+import axios, {AxiosStatic} from 'axios';
+import {extractAxiosErrorMessage} from '../utils';
+
+export class Auth {
+    private axios: AxiosStatic;
+    private apiUrl: string;
+    private clientName: string;
+    private clientVersion: string;
+
+    public static client(apiUrl: string, clientName: string, clientVersion: string) {
+        return new Auth(axios, apiUrl, clientName, clientVersion);
+    }
+
+    constructor(axios: AxiosStatic, apiUrl: string, clientName: string, clientVersion: string) {
+        this.axios = axios;
+        this.apiUrl = apiUrl;
+        this.clientName = clientName;
+        this.clientVersion = clientVersion;
+    }
+
+    register(
+        name: string,
+        lastname: string,
+        email: string,
+        password: string,
+        mnemonic: string,
+        salt: string,
+        privateKey: string,
+        publicKey: string,
+        revocationKey: string,
+        referral?: string,
+        referrer?: string,
+    ) {
+        return this.axios
```

Comment:
Consider enforcing here the same checks that are in the server (easier to develop, therefore). No empty email, password, mnemonic... As this unifies the register, could be a good idea to enforce here the email validation as @internxt/lib [does](https://github.com/internxt/lib/blob/master/src/auth/isValidEmail.ts). What do you think @carlossalasamper?

Also test that the values that you provide in the function (`jest.spy(), expect(spy).toBeCalledWith`...) call are the same that are used to call `this.axios.post` to enforce behavior, just in case any other developer could add something between, introducing a bug.